src/anonymize_proxy.js

```diff
@@ -86,3 +86,19 @@ export const closeAnonymizedProxy = (anonymizedProxyUrl, closeConnections, callb
         });
     return nodeify(promise, callback);
 };
+
+/**
+ * Add a callback on 'tunnelConnectResponded' Event in order to get headers from CONNECT tunnel to proxy
+ * Useful for some proxies that are using headers to send information like ProxyMesh
+ * @param anonymizedProxyUrl
+ * @param tunnelConnectRespondedCallback
+ */
+export const listenConnectAnonymizedProxy = (anonymizedProxyUrl, tunnelConnectRespondedCallback) => {
+    const server = anonymizedProxyUrlToServer[anonymizedProxyUrl];
+    if (!server) {
+        tunnelConnectRespondedCallback(false);
```

Comment:
Would be better to return `null`, to avoid mixing of types (object vs boolean). And you probably need to return from the function.